wgpu-core/src/validation.rs

```diff
@@ -321,6 +333,22 @@ pub enum StageError {
         var: InterfaceVar,
         limit: u32,
     },
+    #[error("Mesh shaders are limited to {limit} output vertices, but the shader has a maximum number of {value}")]
+    TooManyMeshVertices { limit: u32, value: u32 },
+    #[error("Mesh shaders are limited to {limit} output primitives, but the shader has a maximum number of {value}")]
+    TooManyMeshPrimitives { limit: u32, value: u32 },
+    #[error("Mesh or task shaders are limited to {limit} bytes of task payload, but the shader has a task payload of size {value}")]
+    TaskPayloadTooLarge { limit: u32, value: u32 },
+    #[error("Mesh shader's task payload has size {shader:?}, which doesn't match input from previous stage {input:?}")]
+    TaskPayloadMustMatch {
+        input: Option<u32>,
+        shader: Option<u32>,
+    },
+    #[error("Primitive index can only be used in a fragment shader if the preceding shader was a vertex shader or a mesh shader that writes to primitive index.
+    If a mesh shader writes to primitive index, it must be read by the fragment shader.")]
+    PrimitiveIndexError,
+    #[error("Draw id can only be used in a mesh shader if the pipeline has no task shader.")]
```

Comment:
```suggestion
    #[error("DrawId cannot be used in the same pipeline as a task shader")]
```

Avoids the double negative.